### PR TITLE
Deploy shopping cart on centralpark2

### LINKED ISSUE #16: Add an automated deployment test
Deployment to Central Park should be part of the CI build. Since the primary purpose of this repository is to demonstrate deployment to OpenShift, we need to ensure there aren't any regressions and the process documented in [the guide](https://developer.lightbend.com/guides/openshift-deployment/) works through updates.

### PR DESCRIPTION
**Pull request by ignasi35**
Fixes #16

### CODE REVIEW

File: shopping-cart/deploy/scripts/common/installers.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            tar --strip 1 -xvzf openshift-origin-client-tools-v3.11.0-0cbc58b-linux-64bit.tar.gz && chmod +x oc && sudo cp oc /usr/local/bin/ && rm oc
+        )
+        rm -rf tmp/
+    else 
+        echo "! oc already installed."
+    fi
+    oc version && echo "oc CLI installed successfully"
+}
+
+installKustomize() {
```
Comment (ihostage) [inline shopping-cart/deploy/scripts/common/installers.sh:21]: Kubectl 1.14+ have `kustomize` out-of-the-box. We played with him. OC until haven't?

Comment (ignasi35) [reply shopping-cart/deploy/scripts/common/installers.sh:21]: Interesting. 
TBH, I added this code here but it's not used and we'll probably remove it. We've found other ways to manage the YAML files without kustomize. This is a left-over from lagom/lagom-scala-openshift-smoketests

Comment (ihostage) [reply shopping-cart/deploy/scripts/common/installers.sh:21]: > We've found other ways to manage the YAML files without kustomize

We too 😄 
Now for Lagom 1.5.X we migrated from LBO to HELM charts.

Comment (ignasi35) [reply shopping-cart/deploy/scripts/common/installers.sh:21]: Anyway, I think `oc` `3.11` uses `kubectl` `1.11` :-/

**Comment by ignasi35** [comment]
building the docker image in maven fails on travis:

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.example.shoppingcart.impl.ShoppingCartEntityTest
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.908 sec
Results :
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ shopping-cart ---
[INFO] Building jar: /home/travis/build/lagom/lagom-samples/shopping-cart/shopping-cart-java/shopping-cart/target/shopping-cart-1.0-SNAPSHOT.jar
[INFO] 
[INFO] --- docker-maven-plugin:0.26.1:build (build-docker-image) @ shopping-cart ---
[ERROR] DOCKER> Unable to pull 'adoptopenjdk/openjdk8:latest' from registry 'docker-registry-default.centralpark2.lightbend.com' : unauthorized: authentication required (Internal Server Error: 500) [unauthorized: authentication required (Internal Server Error: 500)]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 

```
I've managed to reproduce locally by removing all images from the registries I was logged to:

```
docker images --format='{{println .ID}}'| xargs docker rmi -f  | wc -l 
```

I think the problem is that `fabric8`'s `docker-maven-plugin` [reads](https://dmp.fabric8.io/#registry) the `DOCKER_REGISTRY` env var that we use on the scripts. Another cause for the problem could be that centralpark's registry is set to not fetch images it doesn't contain.

**Comment by TimMoore** [comment]
This will resolve #16, right?

File: shopping-cart/deploy/scripts/common/installers.sh (hunk 1 above)
Comment (ignasi35) [reply shopping-cart/deploy/scripts/common/installers.sh:21]: I realised over the weekend that `kustomize` being part of `kubectl` will require us changing some [docs](https://www.lagomframework.com/documentation/current/java/Migration15.html#Docker-images-and-deployment-specs). Thanks for sharing @ihostage ! ❤️

**Comment by ignasi35** [comment]
> This will resolve #16, right?

yes, I've updated the description

**Comment by ignasi35** [comment]
Me not pressing `send` to publish the https://github.com/lagom/lagom-samples/pull/40#issuecomment-493672554 and also me squashing commits left the history on this PR a bit confusing. Here's the current status:

 - This PR already solves the `Unable to pull 'adoptopenjdk/openjdk8:latest' from registry` [issue](https://github.com/lagom/lagom-samples/pull/40#issuecomment-493672554). The bug was caused by our scripts declaring the `DOCKER_REGISTRY` ENV_VAR which causes `docker-maven-plugin` to behave differently.

At this point, the only issue pending solution is the installation of Kafka/Strimzi on CentralPark2.

File: shopping-cart/deploy/scripts/builds.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+build() {
+    SHOPPING_CART_SOURCES=$1
+    BUILD_TOOL=$2
+
+    if [ "$BUILD_TOOL" == "sbt" ]
+    then
+        (
+            cd $SHOPPING_CART_SOURCES
+            buildSbt $SHOPPING_CART_SOURCES
```
Comment (octonato) [inline shopping-cart/deploy/scripts/builds.sh]:
```suggestion
            sbt clean docker:publishLocal 
```

We can simply inline it here.

File: shopping-cart/deploy/scripts/builds.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        )
+    elif [ "$BUILD_TOOL" == "maven" ]
+    then
+        (
+            cd $SHOPPING_CART_SOURCES
+            buildMvn $SHOPPING_CART_SOURCES
```
Comment (octonato) [inline shopping-cart/deploy/scripts/builds.sh]:
```suggestion
            mvn package docker:build
```

File: shopping-cart/deploy/scripts/builds.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    echo "Built images inventory:$IMAGE_TAG"
+    echo " - - - "
+}
+
+buildSbt() {
+    sbt clean docker:publishLocal 
+}
+buildMvn() {
+    mvn package docker:build
+}
```
Comment (octonato) [inline shopping-cart/deploy/scripts/builds.sh]: From line 14 to 19 can be removed if inlined below

File: shopping-cart/deploy/scripts/builds.sh [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+build() {
+    SHOPPING_CART_SOURCES=$1
+    BUILD_TOOL=$2
+
+    if [ "$BUILD_TOOL" == "sbt" ]
+    then
+        (
+            cd $SHOPPING_CART_SOURCES
```
Comment (octonato) [inline shopping-cart/deploy/scripts/builds.sh]: we can `cd` outside the `if`, but inside a `( )` scope block

**Review by octonato — APPROVED** [review]

File: shopping-cart/deploy/scripts/builds.sh (hunk 5 above)
Comment (ignasi35) [reply shopping-cart/deploy/scripts/builds.sh]: I'm not familiar with the `( )` scoping rules so I trust your changes here. :-)

**Pull request merged**